### PR TITLE
fix(sandbox): return volume mounts in sandbox info and list APIs

### PR DESCRIPTION
**Pull request by xiaojunxiang2023**
## Summary

- Expose `volume_mounts` on Cubelet list `Container` responses and thread them through CubeMaster sandbox info/list into CubeAPI `GET /sandboxes` and list payloads.
- Populate `hostPath` when injecting host-dir mounts so clients can see bind-mount source paths after create.
- Extend `SandboxVolumeMount` with optional `hostPath` and `readOnly`; regenerate `openapi.yml` and web TypeScript schema; update SDK models and `cubemastercli info` output.

## Problem

Host mounts are applied at create time via `metadata["host-mount"]`, but sandbox detail/list APIs always returned `volumeMounts: null`. Web, CLI, and SDK callers could not inspect mounts on a running sandbox.

## Test plan

- [x] `go test ./services/cubebox -run TestToGRPCContainerIncludesVolumeMounts` (Cubelet)
- [x] `go test ./pkg/service/sandbox -run 'TestInjectHostDirMounts|TestCollectVolumeMounts'` (CubeMaster)
- [x] `go test ./cmd/cubemastercli/commands/cubebox -run TestPrintSandboxInfoBlockIncludesVolumeMounts` (CLI)
- [x] `cargo test map_volume_mounts` (CubeAPI)

### CODE REVIEW

**Comment by xiaojunxiang2023** [comment]
> ## Review: PR #696 — Expose volume mounts in sandbox info/list APIs
> **Overall:** Well-structured change that consistently threads volume mount data through the entire stack (Cubelet to CubeMaster to CubeAPI to SDKs to CLI to Web). Tests cover the happy path well. A few items below.
> 
> ### 1. openapi.yml copyright header unintentionally removed
> **File:** openapi.yml (lines 1-6, deleted)
> 
> ## The diff removes the copyright header block. This was likely an accidental deletion. Please restore or confirm dliberate.
> ### 2. Exported Rust struct CubeVolumeMount missing doc comment
> **File:** CubeAPI/src/cubemaster/mod.rs:9
> 
> ## A brief doc comment would help distinguish this deserialization target from the serialization-only VolumeMount.
> ### 3. Exported Go types/functions missing godoc comments
> **Files:**
> 
> * CubeMaster/pkg/service/sandbox/types/types.go:529 - VolumeMountInfo
> * CubeMaster/pkg/service/sandbox/volume_mounts.go:567 - collectVolumeMountsFromContainers
> * CubeMaster/pkg/service/sandbox/volume_mounts.go:589 - volumeMountsToContainerInfo
> 
> All exported but undocumented.
> 
> ### 4. CLI tabwriter alignment may produce ragged output
> **File:** CubeMaster/cmd/cubemastercli/commands/cubebox/info.go:127-137
> 
> The 4-column VOLUME_MOUNTS header is aligned by the same tabwriter as surrounding 2-column rows, likely producing uneven columns.
> 
> ### 5. Test coverage gaps: edge cases in mapping functions
> * map_volume_mounts with empty input
> * map_volume_mounts where all entries filter to empty
> * map_volume_mounts with mixed valid + invalid
> * collectVolumeMountsFromContainers with nil container in the slice
> * volumeMountsToContainerInfo with nil mount or empty slice
> 
> ### 6. volumeMountsToContainerInfo defensive nil check (optional)
> **File:** CubeMaster/pkg/service/sandbox/volume_mounts.go:595
> 
> Nil guard is redundant since its caller already filters nils.
> 
> ### 7. map_volume_mounts filter could use a comment
> **File:** CubeAPI/src/services/sandboxes.rs:736
> 
> Filter intentionally preserves partial entries; an inline comment would make this explicit.
> 
> **Summary:** No security concerns - data-propagation-only change. All existing auth/access controls apply

**Fixed**
- ** 1 ** Restored the `openapi.yml` copyright header removed by the OpenAPI export.
- ** 2 ** Added a doc comment on `CubeVolumeMount` (CubeMaster deserialization shape vs request `VolumeMount`).
- ** 3 ** Added godoc on `VolumeMountInfo`, `collectVolumeMountsFromContainers`, and `volumeMountsToContainerInfo`.
- ** 5 ** Added tests for empty `map_volume_mounts` input, all-empty entries filtered out, nil container in the slice, and nil mount in `volumeMountsToContainerInfo`.
- ** 7 ** Documented why the `map_volume_mounts` filter keeps partial name/path entries.

**No code change (by design)**
- ** 4 ** `cubemastercli info` reuses the same tabwriter as the surrounding 2-column rows; 4-column `VOLUME_MOUNTS` may look uneven but remains readable for ops/debug. Happy to polish formatting in a follow-up if maintainers prefer.
- ** 6 ** Keeping the nil guard in `volumeMountsToContainerInfo` — protobuf slices can contain nil entries and callers may grow over time.

**Comment by kinwin-ustc** [comment]
@ls-ggg Does this conflict with our volume work?

**Comment by xiaojunxiang2023** [comment]
> @ls-ggg Does this conflict with our volume work?
@kinwin-ustc 
Could you help move this forward when you have some time? I think this feature is really necessary for operations troubleshooting.

**Comment by ls-ggg** [comment]
Thanks for your pr，We are currently developing volume-related features that may conflict with your PR. We plan to submit this code later this week, so you can rebase your work against it to implement these features.

**Comment by xiaojunxiang2023** [comment]
> Thanks for your pr，We are currently developing volume-related features that may conflict with your PR. We plan to submit this code later this week, so you can rebase your work against it to implement these features.

OK, I'll rebase my work once the volume-related changes are available.

**Comment by ls-ggg** [comment]
Hi! The core volume capabilities have been merged into master. Please rebase on the latest code and implement your changes on top of it

**Comment by xiaojunxiang2023** [comment]
@ls-ggg ok, rebase done

**Comment by ls-ggg** [comment]
@xiaojunxiang2023 Ack! e2e framework landed in v0.6.0. Please add tests under tests/e2e/sdk_compat. Thanks!

File: sdk/python/tests/test_sandbox.py [hunk 1]
```diff
@@ -727,6 +743,21 @@ def test_list_v2_returns_list(self):
...
+                "name": "hostdir-0",
+                "path": "/mnt/ro",
+                "hostPath": "/tmp/ro",
+                "readOnly": True,
+            }],
+        }]
+        with patch("requests.Session.get", return_value=mock_response(data)):
+            result = Sandbox.list_v2(config=make_config())
+
+        assert result[0]["volumeMounts"][0]["path"] == "/mnt/ro"
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply sdk/python/tests/test_sandbox.py]: Fixed in c83e0ea: the test now asserts `hostPath` and `readOnly` in addition to `path`.

File: CubeMaster/pkg/service/sandbox/volume_mounts.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+//
+
+package sandbox
+
+import (
+	cubeboxv1 "github.com/tencentcloud/CubeSandbox/CubeMaster/api/services/cubebox/v1"
+	"github.com/tencentcloud/CubeSandbox/CubeMaster/pkg/service/sandbox/types"
+)
+
+const (
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeMaster/pkg/service/sandbox/volume_mounts.go]: Added a comment in c83e0ea referencing `Cubelet/pkg/constants/labels.go` so readers know to keep these in sync. Introducing a new import dependency just for two string constants would add more coupling than it removes.

File: CubeMaster/pkg/service/sandbox/volume_mounts.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	}
+	out := make([]*cubeboxv1.VolumeMounts, 0)
+	for _, container := range containers {
+		if container == nil || container.GetType() == containerTypeSandbox {
+			continue
+		}
+		for _, mount := range container.GetVolumeMounts() {
+			if mount == nil {
+				continue
+			}
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeMaster/pkg/service/sandbox/volume_mounts.go:30]: test

[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeMaster/pkg/service/sandbox/volume_mounts.go:30]: 已在  里按 (name, containerPath, hostPath, readonly) 去重，避免多 workload container 时重复出现相同 hostdir mounts。修复见 f18ca4c。

File: CubeMaster/pkg/service/sandbox/volume_mounts.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	cubeboxv1 "github.com/tencentcloud/CubeSandbox/CubeMaster/api/services/cubebox/v1"
+	"github.com/tencentcloud/CubeSandbox/CubeMaster/pkg/service/sandbox/types"
+)
+
+// containerTypeSandbox / containerTypeContainer match Cubelet/pkg/constants/labels.go
+// ContainerTypeSandBox and ContainerTypeContainer — keep in sync if those change.
+const (
+	containerTypeSandbox   = "sandbox"
+	containerTypeContainer = "container"
+)
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeMaster/pkg/service/sandbox/volume_mounts.go]: 这处我先保留了本地字符串常量，并在注释里明确它们与  对齐。考虑到直接引入 constants 包会扩大模块依赖/耦合面，所以先用注释保证一致性，避免后续因依赖图变化引入额外风险。

File: CubeMaster/pkg/service/sandbox/volume_mounts.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeMaster/pkg/service/sandbox/volume_mounts.go:30]: 已在 volumeMountsToContainerInfo 内按 (name, containerPath, hostPath, readonly) 去重，避免多 workload container 时出现重复 hostdir mounts。修复见 f18ca4c。

File: CubeMaster/pkg/service/sandbox/volume_mounts.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeMaster/pkg/service/sandbox/volume_mounts.go]: 这处我保留本地字符串常量，并在注释里注明与 Cubelet/pkg/constants/labels.go 一致。当前不直接引入 constants 包是为了避免额外模块/耦合。

File: CubeMaster/pkg/service/sandbox/volume_mounts.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeMaster/pkg/service/sandbox/volume_mounts.go:30]: Ignore the temporary message 'test' from earlier; the real reply is the one just above (dedup fix f18ca4c).

File: tests/e2e/sdk_compat/cases/auth/test_api_key.py [hunk 5]
```diff
@@ -1,109 +0,0 @@
-# Copyright (c) 2026 Tencent Inc.
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply tests/e2e/sdk_compat/cases/auth/test_api_key.py:1]: Good catch. I restored the unrelated sdk_compat auth coverage and the auth-header behavior in template alias tests, then force-pushed the fix in bb6023f.

File: CubeAPI/src/services/sandboxes.rs [hunk 6]
```diff
@@ -146,7 +146,7 @@ impl SandboxService {
             disk_size_mb: Some(d.disk_size_mb),
             metadata: optional_metadata(d.labels),
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeAPI/src/services/sandboxes.rs:147]: Added an explicit regression test for the host_path-only case and kept the filter behavior aligned with that expectation. Force-pushed in bb6023f.

File: CubeMaster/pkg/service/sandbox/volume_mounts.go [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+	}
+	out := make([]*types.VolumeMountInfo, 0, len(mounts))
+	// A sandbox may have multiple workload containers; hostdir mounts can be injected into each
+	// container, which would lead to duplicate volumeMounts entries in API responses.
+	seen := make(map[string]struct{}, len(mounts))
+	for _, mount := range mounts {
+		if mount == nil {
+			continue
+		}
+		key := mount.GetName() + "\x1f" +
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeMaster/pkg/service/sandbox/volume_mounts.go:70]: Agreed this is a theoretical edge case. For the current hostdir-injection path the separator is fine, so I am leaving the key as-is for now rather than reshaping the helper in this PR.

File: CubeMaster/pkg/service/sandbox/volume_mounts.go [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+import (
+	cubeboxv1 "github.com/tencentcloud/CubeSandbox/CubeMaster/api/services/cubebox/v1"
+	"github.com/tencentcloud/CubeSandbox/CubeMaster/pkg/service/sandbox/types"
+	"strconv"
+)
+
+// containerTypeSandbox / containerTypeContainer match Cubelet/pkg/constants/labels.go
+// ContainerTypeSandBox and ContainerTypeContainer — keep in sync if those change.
+const (
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeMaster/pkg/service/sandbox/volume_mounts.go]: Keeping the local constants with the sync comment for now. Importing Cubelet constants here pulls in a heavier dependency graph than this helper needs, so I prefer not to expand the module surface in this PR.

**Comment by xiaojunxiang2023** [comment]
@ls-ggg Thanks! Added sdk_compat e2e coverage under cases/host-mount/: create with host mounts, then assert get_info and list expose volumeMounts (hostPath/path/readOnly). Rebased onto latest master and force-pushed.

**Comment by ls-ggg** [comment]
@xiaojunxiang2023 
Sorry about that—could you resolve these conflicts? I'm prioritizing this PR today

**Comment by xiaojunxiang2023** [comment]
@ls-ggg Rebased onto latest master and resolved the conflicts (kept master readOnly bool semantics and added hostPath). Force-pushed in f2c99ca.

File: openapi.yml [hunk 9]
```diff
@@ -1374,6 +1374,10 @@ components:
       - name
       - path
       properties:
+        hostPath:
+          type:
+          - string
+          - 'null'
         name:
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply openapi.yml:1377]: Fixed: hostPath is now a plain string in openapi/schema to match Option omit semantics. Force-pushed in f2c99ca.

File: CubeAPI/src/services/sandboxes.rs [hunk 10]
```diff
@@ -1089,6 +1118,50 @@ mod tests {
...
+            container_path: "/mnt/data".to_string(),
+            host_path: "/tmp/data".to_string(),
+            readonly: true,
+        }])
+        .expect("mounts should map");
+
+        assert_eq!(mapped.len(), 1);
+        assert_eq!(mapped[0].name, "hostdir-0");
+        assert_eq!(mapped[0].path, "/mnt/data");
+        assert_eq!(mapped[0].host_path.as_deref(), Some("/tmp/data"));
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeAPI/src/services/sandboxes.rs]: Keeping the current test name. It documents the concrete hostPath-only fixture that previously confused readers; the filter itself remains the general empty-name/path guard.

File: CubeMaster/pkg/service/sandbox/volume_mounts.go [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+package sandbox
+
+import (
+	cubeboxv1 "github.com/tencentcloud/CubeSandbox/CubeMaster/api/services/cubebox/v1"
+	"github.com/tencentcloud/CubeSandbox/CubeMaster/pkg/service/sandbox/types"
+	"strconv"
+)
+
+// containerTypeSandbox / containerTypeContainer match Cubelet/pkg/constants/labels.go
+// ContainerTypeSandBox and ContainerTypeContainer — keep in sync if those change.
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeMaster/pkg/service/sandbox/volume_mounts.go]: Same as before: keeping local constants with the sync comment. Importing Cubelet constants here expands the module graph more than this helper needs.

File: CubeAPI/src/cubemaster/mod.rs [hunk 12]
```diff
@@ -1178,6 +1195,7 @@ pub struct SandboxDetail {
     pub disk_size_mb: i32,
     pub annotations: HashMap<String, String>,
     pub labels: HashMap<String, String>,
+    pub volume_mounts: Vec<CubeVolumeMount>,
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeAPI/src/cubemaster/mod.rs:1196]: Leaving SandboxDetail as-is for now. It is only constructed programmatically from GetSandboxDataItem in this PR, so I am skipping the defensive serde(default) change to avoid another re-review cycle.

**Comment by ls-ggg** [comment]
## PR #696 — Issues (concise)

### Blocking
1. **`volumeMounts` empty on common path** — `collectVolumeMountsFromContainers` skips `type=sandbox`, but template sandboxes often have only that container. Host mounts live there → info/list return `volumeMounts: null`. Live e2e: `test_get_info_returns_volume_mounts` / `test_list_returns_volume_mounts` fail. Unit tests only cover sandbox+workload.

2. **E2E vs API serialization** — Tests assert `readOnly is False`, but CubeAPI omits `readOnly` when false. Will fail after (1) is fixed. Use `bool(m.get("readOnly")) is False`.

### Product / API & SDK surface
3. **`hostPath` should not be user-facing** — Node-local path; not useful (and risky) for clients. Prefer omit from public CubeAPI responses and SDKs; keep `path` / `readOnly` / `name`. This PR currently exposes it in the API, Go `VolumeMount.HostPath`, and Python via raw `volumeMounts` JSON.

4. **SDK field gaps / inconsistency**
   - **Create vs query:** Python already has create-time `VolumeMount(..., read_only=True)` (#1098), but **no typed query fields** on `SandboxInfo` for mounts — only raw dict access (`info.get("volumeMounts")`) if the API returns them.
   - **Go:** `SandboxInfo.VolumeMounts` exists; this PR adds `HostPath` / `ReadOnly` on the struct — `HostPath` should not be part of the public SDK model (see #3).
   - **Python:** no first-class `volume_mounts` / `host_path` attributes; relying on raw API keys is not a stable SDK contract. If mounts are exposed at all, use typed fields aligned with the public API (`name`, `path`, `read_only`) and document them; do not surface `hostPath`.

### Non-blocking / gaps
5. No unit test for **single sandbox container + host mounts**.

### What passed
Unit tests from the PR plan; host-mount create/RW/RO/reject e2e still work (mounts function inside the sandbox; only info/list exposure fails).

---

**Request:** Please share a short **demo of the intended end-to-end effect** after fixes — e.g. create a sandbox with host-mount (RW + RO), then show `GET /sandboxes/:id` and list output (and preferably SDK `get_info` / `list`) with the expected `volumeMounts` shape, so reviewers can confirm what clients are supposed to see.

**Comment by xiaojunxiang2023** [comment]
@ls-ggg Thanks for the clear review — addressed in the latest push (`2899dd4`):

### Blocking
1. **`volumeMounts` empty on common path** — `collectVolumeMountsFromContainers` now collects mounts from **all** containers (including `type=sandbox`), with dedupe by public identity (`name` + `path` + `readOnly`). Added unit coverage for sandbox-only + cross-type dedupe.
2. **E2E `readOnly` omit-false** — updated `test_get_info_returns_volume_mounts` / `test_list_returns_volume_mounts` to use `bool(m.get("readOnly")) is False`.

### Product / API & SDK
3. **`hostPath` removed from public surface** — dropped from CubeMaster `VolumeMountInfo`, CubeAPI `SandboxVolumeMount` / openapi / generated schema, Go `VolumeMount`, CLI `info` output, and e2e/SDK assertions. Public shape is `name` / `path` / `readOnly` only.
4. **SDK typed query fields (Python)** — Go already has typed `SandboxInfo.VolumeMounts` (without `HostPath`). Python still returns mounts via the existing raw/dict `SandboxInfo` path (`info["volumeMounts"]`); adding first-class `volume_mounts` attributes can be a small follow-up if you want that in this PR rather than later.

### Non-blocking
5. Covered by the new sandbox-only unit test above.

### Expected client view (after create with RW+RO host mounts)
```json
"volumeMounts": [
  {"name": "hostdir-0", "path": "/mnt/rw"},
  {"name": "hostdir-1", "path": "/mnt/ro", "readOnly": true}
]
```
(`readOnly` omitted when false; no `hostPath`.)

Unit tests for CubeMaster / CubeAPI `map_volume_mounts` / Go+Python SDK volume-mount decoding are green. Happy to run a live create→info/list demo against a cluster if you want that captured in-thread next.

File: sdk/go/models.go [hunk 13]
```diff
@@ -39,8 +39,9 @@ type SandboxInfo struct {
 }
 
 type VolumeMount struct {
-	Name string `json:"name"`
-	Path string `json:"path"`
+	Name     string `json:"name"`
+	Path     string `json:"path"`
+	ReadOnly *bool  `json:"readOnly,omitempty"`
 }
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply sdk/go/models.go:46]: Agreed — aligned `VolumeMount.ReadOnly` to a plain `bool` so omitted/`false` both decode as `false`, matching CubeAPI omit-false and the other layers.

File: CubeMaster/pkg/service/sandbox/volume_mounts.go [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+	return out
+}
+
+// volumeMountsToContainerInfo converts cubelet mounts into API-facing JSON structs.
+// HostPath is intentionally omitted: node-local paths are not part of the public
+// sandbox info/list contract.
+func volumeMountsToContainerInfo(mounts []*cubeboxv1.VolumeMounts) []*types.VolumeMountInfo {
+	if len(mounts) == 0 {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeMaster/pkg/service/sandbox/volume_mounts.go:58]: Good catch — documented the hostdir-oriented caveat on the dedup key so future reuse for other mount types does not silently collapse distinct entries.

File: CubeAPI/src/services/sandboxes.rs [hunk 15]
```diff
@@ -146,7 +146,7 @@ impl SandboxService {
             disk_size_mb: Some(d.disk_size_mb),
             metadata: optional_metadata(d.labels),
             state: sandbox_state_from_status(d.status),
-            volume_mounts: None,
+            volume_mounts: map_volume_mounts(&d.volume_mounts),
```
[the remark this answers is not in the document]
Comment (xiaojunxiang2023) [reply CubeAPI/src/services/sandboxes.rs:149]: Thanks for flagging — on the current HEAD (`576e22a`) `get_sandbox()` already maps mounts:

```rust
volume_mounts: map_volume_mounts(&d.volume_mounts),
```

(see `CubeAPI/src/services/sandboxes.rs` around the `SandboxDetail` construction). Cubemaster `into_first_sandbox` also forwards `item.volume_mounts`. Happy to re-check if you're looking at a different revision.

**Comment by ls-ggg** [comment]
## Retest update (`576e22a`)

Thanks for the follow-up — the earlier issues look fixed on a live deploy:

- `volumeMounts` are returned for the common single `type=sandbox` template path
- public responses omit `hostPath`
- e2e `readOnly` assertions match CubeAPI’s omit-when-false serialization
- `cases/host-mount/test_create_with_mount.py` — **6/6 passed**

### Remaining issue: rootfs mount leaks into public `volumeMounts`

After collecting mounts from all containers (including `type=sandbox`), info/list now also expose the internal writable-rootfs mount, e.g.:

```json
[
  { "name": "cube_rootfs_rw", "path": "/" },
  { "name": "hostdir-0", "path": "/mnt/rw" },
  { "name": "hostdir-1", "path": "/mnt/ro", "readOnly": true }
]
```

Hostdir entries look correct, but `cube_rootfs_rw` / `path: "/"` is an implementation detail and probably should not be part of the public sandbox info/list contract (or the SDK surface).

**Suggestion:** keep collecting from sandbox containers (needed for host-mount), but filter out rootfs-style mounts — e.g. `container_path == "/"` and/or known names like `cube_rootfs_*` — so clients only see user-facing mounts (`name` / `path` / `readOnly`).

**Comment by xiaojunxiang2023** [comment]
@ls-ggg Thanks for the retest — fixed the rootfs leak in the latest push.

Public `volumeMounts` now filters internal rootfs mounts (`container_path == "/"` and/or names prefixed with `cube_rootfs_`) while still collecting hostdir mounts from `type=sandbox` containers. Added unit coverage and tightened the host-mount e2e assertions so `cube_rootfs_*` / `path: "/"` must not appear.

Expected shape after RW+RO host mounts:

```json
[
  {"name": "hostdir-0", "path": "/mnt/rw"},
  {"name": "hostdir-1", "path": "/mnt/ro", "readOnly": true}
]
```

**Comment by ls-ggg** [comment]
## Retest (`f5bcd223`) — LGTM

Live deploy of CubeAPI / CubeMaster / Cubelet on the test host; previous issues look fixed.

### Fixed / verified
- `volumeMounts` returned for the common single `type=sandbox` template path
- Public info/list omit `hostPath`
- Internal rootfs mounts (`cube_rootfs_*` / `path: "/"`) filtered out
- e2e `readOnly` assertions match omit-when-false serialization

### Observed public shape
```json
[
  { "name": "hostdir-0", "path": "/mnt/rw" },
  { "name": "hostdir-1", "path": "/mnt/ro", "readOnly": true }
]
```

### Tests
- PR unit tests (Cubelet / CubeMaster / CLI / CubeAPI `map_volume_mounts`) — passed
- `cases/host-mount/test_create_with_mount.py` — **6/6 passed**

**LGTM**

**Comment by xiaojunxiang2023** [comment]
@ls-ggg Thanks for the thorough retest and LGTM — glad the public `volumeMounts` shape looks right on the live deploy.

**Comment by lkml-likexu** [comment]
It looks like we need:

```
diff --git a/Cubelet/doc/cubelet-api.md b/Cubelet/doc/cubelet-api.md
index ab14260..f87e6b3 100644
--- a/Cubelet/doc/cubelet-api.md
+++ b/Cubelet/doc/cubelet-api.md
@@ -433,6 +433,7 @@ Capability contains the container capabilities to add or drop
 | finished_at | [int64](#int64) |  | exit time of the container in nanoseconds. |
 | labels | [Container.LabelsEntry](#cubelet-services-cubebox-v1-Container-LabelsEntry) | repeated |  |
 | paused_at | [int64](#int64) |  |  |
+| volume_mounts | [VolumeMounts](#cubelet-services-cubebox-v1-VolumeMounts) | repeated | Mounts declared on the container spec (including host-dir mounts). |

```

**Comment by xiaojunxiang2023** [comment]
@lkml-likexu Thanks for catching this — the `Container` table in `Cubelet/doc/cubelet-api.md` was missing `volume_mounts` even though the proto already exposes it.

Opened a follow-up docs PR with the exact row you suggested: https://github.com/TencentCloud/CubeSandbox/pull/1241
